spec/defines/dropin_file_spec.rb

```diff
@@ -3,187 +3,182 @@
 require 'spec_helper'
 
 describe 'systemd::dropin_file' do
-  context 'supported operating systems' do
-    on_supported_os.each do |os, facts|
-      context "on #{os}" do
-        let(:facts) { facts }
-
-        let(:title) { 'test.conf' }
-
-        let(:params) do
-          {
-            unit: 'test.service',
-            content: 'random stuff',
-          }
-        end
-
-        it { is_expected.to compile.with_all_deps }
-
-        it {
-          expect(subject).to create_file("/etc/systemd/system/#{params[:unit]}.d").with(
-            ensure: 'directory',
-            recurse: 'true',
-            purge: 'true',
-            selinux_ignore_defaults: false
-          )
-        }
+  _, facts = on_supported_os.first
```

Comment:
Perhaps you can introduce a helper method `default_os_facts`. I also wouldn't mind knowing which OS is being tested on.